### PR TITLE
ROX-20160: Add Go build to main in Konflux

### PR DESCRIPTION
**Pull request by msugakov**
## Description

Onboarding the Go binary builds into the main image build on Konflux.

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

### Here I tell how I validated my change

Konflux build succeeds.

Running image `quay.io/redhat-user-workloads/rh-acs-tenant/acs/main:on-pr-e7a2cd4e454aecc8b6f3e0f5e2a09ab3ff2b9345` with Docker/Podman locally and testing that all binaries can be executed (failures expected due to missing configuration files). 

### Reminder for reviewers

In addition to reviewing code here, reviewers **must** also review testing and request further testing in case the
performed one does not seem sufficient. As a reviewer, you must not approve the change until you understand the
performed testing and you are satisfied with it.

### CODE REVIEW

File: image/rhel/konflux.Dockerfile [hunk 1]
```diff
@@ -52,6 +60,44 @@ COPY .konflux/rocksdb ./
...
+COPY --from=rpm-installer "$GO_BUILDER_STAGE_PATH" /
+
+COPY --from=rocksdb-builder /staging/librocksdb.a /lib/rocksdb/librocksdb.a
+COPY --from=rocksdb-builder /staging/include /lib/rocksdb/include
+
+WORKDIR /go/src/github.com/stackrox/rox/app
+
+COPY . .
+
+# TODO: remove me
```
Comment (tommartensen) [inline image/rhel/konflux.Dockerfile:75]: Possible workaround: https://github.com/stackrox/stackrox/blob/master/scripts/konflux/fail-build-if-git-is-dirty.sh#L17-L19.

File: image/rhel/konflux.Dockerfile [hunk 2]
```diff
@@ -81,9 +127,20 @@ COPY --from=rpm-installer "$FINAL_STAGE_PATH" /
...
+COPY --from=go-builder /go/src/github.com/stackrox/rox/app/image/rhel/bin/kubernetes-sensor /stackrox/bin/
+COPY --from=go-builder /go/src/github.com/stackrox/rox/app/image/rhel/bin/sensor-upgrader /stackrox/bin/
+COPY --from=go-builder /go/src/github.com/stackrox/rox/app/image/rhel/bin/admission-control /stackrox/bin/
+COPY --from=go-builder /go/src/github.com/stackrox/rox/app/image/rhel/static-bin/* /stackrox/
+RUN GOARCH=$(uname -m) ; \
+    case $GOARCH in x86_64) GOARCH=amd64 ;; aarch64) GOARCH=arm64 ;; esac ; \
+    ln -s /assets/downloads/cli/roxctl-linux-$GOARCH /stackrox/roxctl ; \
+    ln -s /assets/downloads/cli/roxctl-linux-$GOARCH /assets/downloads/cli/roxctl-linux
+
+# TODO: rhacs-main/Dockerfile.in?ref_type=heads#L107-122
```
Comment (tommartensen) [inline image/rhel/konflux.Dockerfile:143]: As discussed this is to be addressed in a separate PR.
While looking at https://github.com/stackrox/stackrox/blob/master/image/rhel/fetch-stackrox-data.sh & https://gitlab.cee.redhat.com/stackrox/rhacs-midstream/-/blob/rhacs-1.0-rhel-8/distgit/containers/rhacs-main/pre-build-script?ref_type=heads, I was wondering why we do not include the Istio CVE list in upstream images.

Comment (msugakov) [reply image/rhel/konflux.Dockerfile:143]: Looks like it was removed upstream about a year ago. See https://github.com/stackrox/stackrox/pull/4341

Probably downstream was forgotten.

Asked in Slack https://redhat-internal.slack.com/archives/C033SKHG9CZ/p1712598818791699

File: image/rhel/konflux.Dockerfile (hunk 1 above)
Comment (msugakov) [reply image/rhel/konflux.Dockerfile:75]: I felt that git-restoring any `package-lock.json` files in Konflux context is a bit too custom for `fail-build-if-git-is-dirty.sh`, and besides it may break UI build if `fail-build-if-git-is-dirty.sh` is called at UI build stage (no case right now but who knows).

My suggestion would be to just remove this comment and this `git diff`. I am just not sure if that's really Konflux changing it and what's the nature of the changes. I wanted to look at logs for this. Unfortunately, logs for the latest build are gone at the moment.

File: .tekton/main-pull-request.yaml [hunk 3]
```diff
@@ -87,6 +87,10 @@ spec:
         requests:
           cpu: 1
 
+  # The pipeline regularly takes >1h to finish.
```
[the remark this answers is not in the document]
Comment (tommartensen) [reply .tekton/main-pull-request.yaml:90]: Yes, I had some builds time out after 1h.

File: image/rhel/konflux.Dockerfile [hunk 4]
```diff
@@ -173,4 +173,4 @@ ENV PATH="/stackrox:$PATH" \
 
 USER 4000:4000
 
-# TODO: rhacs-main/Dockerfile.in?ref_type=heads#L127
+HEALTHCHECK CMD curl --insecure --fail https://127.0.0.1:8443/v1/ping
```
[the remark this answers is not in the document]
Comment (tommartensen) [reply image/rhel/konflux.Dockerfile]: removed here. Should we also remove this upstream & downstream?

**Comment by tommartensen** [comment]
Follow-up for EC failing due to `facebook/rocksdb`: https://issues.redhat.com/browse/ROX-23545

File: image/rhel/konflux.Dockerfile (hunk 4 above)
[the remark this answers is not in the document]
Comment (msugakov) [reply image/rhel/konflux.Dockerfile]: Yes, it should be useful to clean things up. Started a thread to see if there are points to keep any of them https://redhat-internal.slack.com/archives/CELUQKESC/p1712665310487539

**Review by msugakov — COMMENTED** [review]
LGTM. Code-wise it's good to go.
Two things:
1. Konflux CI looks red. It's worth to check and re-run, maybe failures aren't related.
2. Since I'm the original creator of this PR, I can't approve it. You'll have to approve it yourself to be able to merge.

**Comment by tommartensen** [comment]
> Konflux CI looks red. It's worth to check and re-run, maybe failures aren't related.

operator build failed with one-off (?) OOM during `prefetch-dependencies`. Re-run succeeded.
EC fail because of known issues with CVEs + rocksdb.

**Review by tommartensen — APPROVED** [review]
